lib/lite_rolling_switch.dart

```diff
@@ -86,25 +86,25 @@ class _RollingSwitchState extends State<LiteRollingSwitch>
       });
     });
     turnState = widget.value;
-    _determine();
+    _determine(changeState: turnState);
   }
 
   @override
   Widget build(BuildContext context) {
-    Color transitionColor = Color.lerp(widget.colorOff, widget.colorOn, value);
+    Color? transitionColor = Color.lerp(widget.colorOff, widget.colorOn, value);
 
     return GestureDetector(
       onDoubleTap: () {
         _action();
-        if (widget.onDoubleTap != null) widget.onDoubleTap();
+        widget.onDoubleTap!();
       },
       onTap: () {
         _action();
-        if (widget.onTap != null) widget.onTap();
+        widget.onTap!();
```

Comment:
This could result in error as the onTap callback could be null. So, you should probably have the null check before calling it.